packages/mobile/src/account/profileInfo.ts

```diff
@@ -90,33 +91,35 @@ export function* uploadNameAndPicture() {
 }
 
 // this function gives permission to the recipient to view the user's profile info
-export function* giveProfileAccess(recipientAddresses: string[]) {
+export function* giveProfileAccess(walletAddress: string) {
+  // TODO: check if key for recipient already exists, skip if yes
   try {
-    // TODO: check if key for recipient already exists, skip if yes
+    const walletToAccountAddress: WalletToAccountAddressType = yield select(
+      walletToAccountAddressSelector
+    )
+    const accountAddress = walletToAccountAddress[walletAddress] ?? walletAddress
```

Comment:
Should we add an endpoint to `blockchain-api` to do this `walletAddress -> accountAddress` lookup in case we don't have the mapping in `walletToAccountAddress`?
Exposing what is done in the transaction feed to get it.